docs/Migrating_LLB_to_a_PPL_based_Custom_Device.md

```diff
@@ -0,0 +1,203 @@
+
+#### Incompatible LabVIEW Features
+
+Some LabVIEW features are not compatible with PPL. For example, public malleable VIs (VIMs) cannot be included in a PPL for export. Only VIMs set with a private scope can be included. To resolve this issue, remove these features from your custom device at the code or project level.
+
+#### RT Driver VI Path Errors
+
+Opening a PPL based custom device in VeriStand will generate invalid path errors for the RT Driver VIs. VeriStand System Explorer does not recognize PPL paths. These errors do not affect how the Custom Device runs. To resolve this issue, refer to the [Update VeriStand System Explorer](https://niveristand-custom-device-handbook.readthedocs.io/en/latest/Migrating_LLB_to_a_PPL_based_Custom_Device.html#update-veristand-system-explorer).
+
+#### Deployment Errors
+
+Some custom device APIs, such as the NI VeriStand Custom Device Channel APIs, use Global Data References. This will cause deployment errors for a PPL based custom device. To resolve this issue, refer to the [Update Global Data References](https://niveristand-custom-device-handbook.readthedocs.io/en/latest/Migrating_LLB_to_a_PPL_based_Custom_Device.html#update-global-data-references).
+
+### Implementing a PPL Based Custom Device
+
+To create a new PPL based Custom Device, use the [VeriStand Custom Device Wizard](https://github.com/ni/niveristand-custom-device-wizard/releases) to generate a template project.
+
+### Migrating an LLB Based Custom Device to PPL
+
+To migrate to a PPL based custom device, you must perform the following actions.
+
+1. Update VeriStand System Explorer
+2. Update Global Data References
+3. Update Libraries
+4. Update the XML
+5. Build the PPL File
+
+#### Update VeriStand System Explorer
+
+To remove RT Driver path errors, update the *Initialization* and *Action on Compile* VIs.
+
+Inside the Initialization VI, add a *Clear RT Drivers* subVI to delete all RT Driver paths from the system definition after the custom device is created. The paths can be removed because they are only used during and after deployment.
+
+The following image displays where to insert the subVI.
+
+![](images/Initialization_Change.png)
+
+Inside the Action on Compile VI, add a *Set RT Driver* subVI to re-insert the corresponding RT Driver paths before system deployment. VeriStand keeps a copy of the deployed system definition in the local cache unless the system definition is unmodified between deployments. The latest changes are preserved, allowing VeriStand to use the "complete" system definition from the local cache during deployment.
+
+The following image displays where to insert the subVI.
+
+![](images/Action_on_Compile.png)
+
+For an example of these subVI structures, refer to the [FPGA Add-on Custom Device](https://github.com/ni/niveristand-fpga-addon-custom-device). This custom device will be available in the upcoming version of the NI VeriStand Custom Device Wizard.
```

Comment:
We should remove this section which references the FPGA Add on as an example, as the PR containing these changes was abandoned.